pkg/postage/interface.go

```diff
@@ -97,4 +99,7 @@ type BatchEventListener interface {
 	HandleCreate(*Batch, *big.Int) error
 	HandleTopUp(id []byte, newBalance *big.Int)
 	HandleDepthIncrease(id []byte, newDepth uint8)
+	HandleStampExpiry(id []byte)
```

Comment:
This is actually not correct. Because of this change we had to change the BatchListener which is a different interface.

I think the better option here is to define a new interface and not function.

```go
type BatchExpiryHandler interface {
    HandleStampExpiry([]byte)
}
```

In batch store instead of passing in ExpiredFn you will pass in this interface. And in `cleanup` you would call this `HandleStampExpiry` function using the interface if it is Set by the Setter. As an example you can look at the RadiusSetter defined above in this file.